CMakeLists.txt

```
@@ -0,0 +1,822 @@
+
+if(GIT_DESC)
+  string(REGEX REPLACE "^v([0-9]+)\\..*" "\\1" VERSION_MAJOR "${GIT_DESC}")
+  string(REGEX REPLACE "^v[0-9]+\\.([0-9]+).*" "\\1" VERSION_MINOR
+                       "${GIT_DESC}")
+  string(REGEX REPLACE "^v[0-9]+\\.[0-9]+\\.([0-9]+).*" "\\1" VERSION_PATCH
+                       "${GIT_DESC}")
+  string(REGEX REPLACE "^v[0-9]+\\.[0-9]+\\.[0-9]+(.*)" "\\1" VERSION_GIT
+                       "${GIT_DESC}")
+
+  git_local_changes(GIT_STATE)
+  if("${GIT_STATE}" STREQUAL "DIRTY")
+    set(VERSION_GIT "${VERSION_GIT}-dirty")
+  endif()
+
+  execute_process(
+    COMMAND git log -1 --format=%ai
+    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+    OUTPUT_VARIABLE GIT_COMMIT_DATE
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+  # take only the date from the git timestamp:
+  string(REGEX REPLACE "^([0-9\\-]+) .*" "\\1" VERSION_DATE
+                       "${GIT_COMMIT_DATE}")
+else()
+  file(STRINGS VERSION VERSION_INFO)
+  foreach(line ${VERSION_INFO})
+    if(${line} MATCHES "^([^#].*)=[ \t]*(.*)$")
+      set(key ${CMAKE_MATCH_1})
+      set(value ${CMAKE_MATCH_2})
+      string(REGEX REPLACE "[ \t\n]+$" "" key "${key}")
+      string(REGEX REPLACE "[ \t\n]+$" "" value "${value}")
+      set(VERSION_${key} "${value}")
+      continue()
+    endif()
+  endforeach()
+endif()
+
+project(
+  cp2k
+  DESCRIPTION "CP2K"
+  LANGUAGES Fortran C CXX)
+set(cp2k_VERSION
+    ${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH}${VERSION_GIT})
+set(cp2k_APIVERSION ${VERSION_MAJOR}.${VERSION_MINOR})
+
+if(NOT DEFINED CMAKE_CUDA_STANDARD)
+  set(CMAKE_CUDA_STANDARD 11)
+  set(CMAKE_CUDA_STANDARD_REQUIRED ON)
+endif()
+
+# set language and standard
+set(CMAKE_CXX_STANDARD 11)
+
+find_package(PkgConfig)
+
+# ##############################################################################
+# Define the paths for static libraries and executables
+# ##############################################################################
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY
+    ${cp2k_BINARY_DIR}/lib
+    CACHE PATH "Single output directory for building all libraries.")
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY
+    ${cp2k_BINARY_DIR}/bin
+    CACHE PATH "Single output directory for building all executables.")
+
+# Search for rocm in common locations
+
+# list(APPEND CMAKE_PREFIX_PATH /opt/rocm/hip /opt/rocm)
+
+foreach(__var ROCM_ROOT CRAY_ROCM_ROOT ORNL_ROCM_ROOT CRAY_ROCM_PREFIX
+              ROCM_PREFIX CRAY_ROCM_DIR)
+  if($ENV{${__var}})
+    list(APPEND CMAKE_PREFIX_PATH $ENV{__var})
+    set(ROCM_PATH
+        $ENV{__var}
+        CACHE PATH "Path to ROCm installation")
+  endif()
+endforeach()
+
+# =================================================================================================
+# OPTIONS
+include(CMakeDependentOption)
+
+# make MPI optional as well. More work but might be useful set(CP2K_USE_MPI ON
+# CACHE BOOL "Enable MPI support")
+
+set(CP2K_SET_POSITION_INDEPENDENT_CODE
+    ON
+    CACHE BOOL "Enable position independent code")
```

I would recommend using `option` instead for all the `BOOL` options here (https://cmake.org/cmake/help/latest/command/option.html?highlight=option)
```suggestion
option(CP2K_SET_POSITION_INDEPENDENT_CODE
    ON
    "Enable position independent code")
```
It basically lets you omit the `CACHE` and `BOOL` (it's otherwise equivalent).